cohortextractor2/dsl.py

```diff
@@ -161,9 +169,109 @@ class PatientSeries:
     def __init__(self, value: Value):
         self.value = value
 
+    @staticmethod
+    def _get_other(other: Expression) -> Comparator | Value:
+        if isinstance(other, PatientSeries):
+            return other.value
+        return other
 
-Expression = Union[PatientSeries, bool, int, str]
+    def __gt__(self, other: Expression) -> Comparator:
+        return self.value > self._get_other(other)
 
+    def __ge__(self, other: Expression) -> Comparator:
+        return self.value >= self._get_other(other)
 
-def categorise(mapping: dict[str, Expression]) -> PatientSeries:
-    """Represents a switch statement."""
+    def __lt__(self, other: Expression) -> Comparator:
+        return self.value < self._get_other(other)
+
+    def __le__(self, other: Expression) -> Comparator:
+        return self.value <= self._get_other(other)
+
+    def __eq__(self, other: Expression) -> Comparator:  # type: ignore[override]
+        return self.value == self._get_other(other)
+
+    def __ne__(self, other: Expression) -> Comparator:  # type: ignore[override]
+        return self.value != self._get_other(other)
+
+    def __and__(self, other: Expression) -> Comparator:
+        return self.value & self._get_other(other)
+
+    def __or__(self, other: Expression) -> Comparator:
+        return self.value | self._get_other(other)
+
+    def __invert__(self) -> Comparator:
+        return ~self.value
+
+    def __repr__(self) -> str:
+        return f"PatientSeries(value={self.value})"
+
+
+Expression = Union[PatientSeries, Comparator]
+
+
+def categorise(mapping: dict[str, Expression], default: bool | None) -> PatientSeries:
```

Comment:
I'd probably spell `bool | None` as `Optional[bool]`, which I think expresses the semantics more clearly.

Reply:
I did that, `black` changed it

Reply:
Same as above, I suppose. I think that's a defect in Black, even if we were using 3.10.

Reply:
Wasn't black, it was pyupgrade, and it's doing it because this file has `from __future__ import annotations` in it